UnifiedApi/Client/Services/SessionService.cs

```diff
@@ -48,55 +52,29 @@ public class SessionService : BaseClientSubmitter<SessionService>
   ///   Ctor to instantiate a new SessionService
   ///   This is an object to send task or get Results from a session
   /// </summary>
-  public SessionService(ILoggerFactory                                  loggerFactory,
-                        Api.gRPC.V1.Submitter.Submitter.SubmitterClient controlPlaneService,
-                        TaskOptions                                     taskOptions = null)
-    : base(loggerFactory)
+  public SessionService(ILoggerFactory          loggerFactory,
+                        ChannelBase             channel,
+                        [CanBeNull] TaskOptions taskOptions = null,
```

Comment:
use `?` instead of `[CanBeNull]`. This might require modifications in the csproj.

Reply:
# 472